src/main/kotlin/no/nav/familie/ef/sak/karakterutskrift/AutomatiskBrevInnhentingKarakterutskriftService.kt

```diff
@@ -0,0 +1,59 @@
+package no.nav.familie.ef.sak.karakterutskrift
+
+import no.nav.familie.ef.sak.infrastruktur.exception.Feil
+import no.nav.familie.ef.sak.oppgave.OppgaveService
+import no.nav.familie.ef.sak.oppgave.OppgaveUtil
+import no.nav.familie.kontrakter.felles.Behandlingstema
+import no.nav.familie.kontrakter.felles.Tema
+import no.nav.familie.kontrakter.felles.oppgave.FinnOppgaveRequest
+import no.nav.familie.prosessering.internal.TaskService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.time.Year
+
+@Service
+class AutomatiskBrevInnhentingKarakterutskriftService(
+    private val taskService: TaskService,
+    private val oppgaveService: OppgaveService,
+) {
+
+    val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    val fristHovedperiode = LocalDate.parse("2023-05-17")
+    val fristutvidet = LocalDate.parse("2023-05-18")
+
+    @Transactional
+    fun opprettTasks(brevtype: KarakterutskriftBrevtype, liveRun: Boolean) {
+        val mappeId = oppgaveService.finnMapper(OppgaveUtil.ENHET_NR_NAY)
+            .single { it.navn == "64 Utdanning" }.id
+
+        val opppgaver = oppgaveService.hentOppgaver(
+            FinnOppgaveRequest(
+                tema = Tema.ENF,
+                behandlingstema = Behandlingstema.Skolepenger,
+                fristFomDato = when (brevtype) {
+                    KarakterutskriftBrevtype.HOVEDPERIODE -> fristHovedperiode
+                    KarakterutskriftBrevtype.UTVIDET -> fristutvidet
+                },
+                fristTomDato = fristHovedperiode,
```

Comment:
Hvorfor er det både fom og tom dato for frist? Og hvorfor er det kun fom som tar hensyn til utvidet?

Reply:
Man kan ikke søke på en spesifikk frist, kun en periode. Men riktig at vi burde tatt hensyn til brevtype også for tomDato 😬

Reply:
Fikset nå ✅